src/app/qgsdecorationitem.cpp

```diff
@@ -65,6 +66,8 @@ void QgsDecorationItem::projectRead()
   QgsDebugMsg( "Entered" );
   mEnabled = QgsProject::instance()->readBoolEntry( mNameConfig, "/Enabled", false );
   mPlacement = static_cast< Placement >( QgsProject::instance()->readNumEntry( mNameConfig, "/Placement", static_cast< int >( mPlacement ) ) );
+  mMarginUnits = static_cast< MarginUnits >( QgsProject::instance()->readNumEntry( mNameConfig, "/MarginUnits", static_cast< int >( mMarginUnits ) ) );
```

Comment:
If mMarginUnit is changed to QgsSymbolV2::OutputUnit then this should become:

    mMarginUnits = QgsSymbolLayerV2Utils::decodeOutputUnit( QgsProject::instance()->readEntry( mNameConfig, "/MarginUnits", QgsSymbolLayerV2Utils::encodeOutputUnit( mMarginUnits )  ) );